tests/lib.rs

```diff
@@ -387,3 +387,30 @@ fn foreach() {
         _ => unreachable!(),
     });
 }
+
+#[test]
+fn retain() {
```

Comment:
I'd like to also see tests for `Fit` and `Reserve`, though I guess that might be hard? Having a test that at least _runs_ the operations and then do an insert+get to see that it doesn't crash would be good though.